hack/lib/ui.bash

```diff
@@ -77,6 +78,29 @@ function debugging.finish {
   exec 19>&-
 }
 
+function await-at-fail.setup {
+  if (( INTERACTIVE )); then
+    logger.info 'Skipping await-at-fail because running as interactive user'
+    return 0
+  fi
+  local labeled pr_api
+  pr_api="https://api.github.com/repos/openshift-knative/serverless-operator/pulls/${PULL_NUMBER:-}"
+  labeled="$(curl -s -L "$pr_api" \
+    | jq -r '.labels[] | select( .name == "ci/await-at-fail" ) .name' \
+    | wc -l)"
+  if (( labeled )); then
+    logger.info 'Adding await at fail because of "ci/await-at-fail" label added'
+    error_handlers.register await-at-fail
+  else
+    logger.info 'Add "ci/await-at-fail" label to PR, if you like to await after failure to allow debugging'
+  fi
+}
+
+function await-at-fail {
+  logger.info 'Awaiting at max 1h for user to debug'
+  timeout 3600 "[[ \$(oc get configmap done -n debug -o name | wc -l) -eq 0 ]]"
+}
```

Comment:
This seems generically useful, it doesn't seem specific to ui tests, can we put these functions in common lib?

I'd call the function and the label `await_on_error` to be consistent with `error_handlers` terminology